pages/dois/DoisHandler.inc.php

```diff
@@ -0,0 +1,100 @@
+
+        $context = $request->getContext();
+        $contextId = $context->getId();
+
+        $enabledDoiTypes = $context->getData(Context::SETTING_ENABLED_DOI_TYPES);
+
+        $templateMgr = TemplateManager::getManager($request);
+
+        $commonArgs = [
+            'doiPrefix' => $context->getData(Context::SETTING_DOI_PREFIX),
+            'doiApiUrl' => $request->getDispatcher()->url($request, PKPApplication::ROUTE_API, $context->getPath(), 'dois'),
+            'lazyLoad' => true,
+            'enabledDoiTypes' => $enabledDoiTypes,
+            'registrationAgencyInfo' => $this->_getRegistrationAgencyInfo($context),
+        ];
+
+        HookRegistry::call('DoisHandler::setListPanelArgs', [&$commonArgs]);
+
+        $stateComponents = [];
+
+        if (!empty($enabledDoiTypes)) {
+            $submissionDoiListPanel = new DoiListPanel(
+                'submissionDoiListPanel',
+                __('plugins.pubIds.doi.manager.monographDois'),
+                array_merge(
+                    $commonArgs,
+                    [
+                        'apiUrl' => $request->getDispatcher()->url($request, PKPApplication::ROUTE_API, $context->getPath(), 'submissions'),
+                        'getParams' => [
+                            'stageIds' => [WORKFLOW_STAGE_ID_PUBLISHED, WORKFLOW_STAGE_ID_PRODUCTION],
+                        ],
+                        'itemType' => 'submission'
+                    ]
+                )
+            );
+            $stateComponents[$submissionDoiListPanel->id] = $submissionDoiListPanel->getConfig();
+        }
+
+        // DOI settings
+        $router = $request->getRouter();
+        $dispatcher = $request->getDispatcher();
+
+        $contextApiUrl = $dispatcher->url($request, PKPApplication::ROUTE_API, $context->getPath(), 'contexts/' . $context->getId());
+        $supportedFormLocales = $context->getSupportedFormLocales();
+        $localeNames = AppLocale::getAllLocales();
+        $locales = array_map(function ($localeKey) use ($localeNames) {
+            return ['key' => $localeKey, 'label' => $localeNames[$localeKey]];
+        }, $supportedFormLocales);
+
+        $doiSetupSettingsForm = new DoiSetupSettingsForm($contextApiUrl, $locales, $context);
+        $doiRegistrationSettingsForm = new PKPDoiRegistrationSettingsForm($contextApiUrl, $locales, $context);
+        $stateComponents[PKPDoiSetupSettingsForm::FORM_DOI_SETUP_SETTINGS] = $doiSetupSettingsForm->getConfig();
+        $stateComponents[PKPDoiRegistrationSettingsForm::FORM_DOI_REGISTRATION_SETTINGS] = $doiRegistrationSettingsForm->getConfig();
+
+        $templateMgr->setState(['components' => $stateComponents]);
+
+        $templateMgr->assign([
+            'pageTitle' => __('plugins.pubIds.doi.manager.displayName'),
+            'displayMonographsTab' => !empty($enabledDoiTypes),
+        ]);
+
+        $templateMgr->display('management/dois.tpl');
+    }
+}
```

Comment:
Not a merge blocker but I think a fair chunk of this can be moved to pkp-lib. Where possible, we can avoid variable names like `$displayMonographsTab` and replace them with `$displaySubmissionsTab`.

Reply:
I went ahead and refactored this now.